tools/dataset_object_processor.py

```diff
@@ -21,6 +22,21 @@
 ROBOT_PATH = ""  # TODO, which dataset is this? Robot fetch?
 
 
+class MemoryUnitConverter:
+    """
+    class to convert computer memory value units, e.g.
+    1,000,000 bytes to 1 megabyte
```

Comment:
1048576? :eyes:

Reply:
Embarrassingly, I just learned that it is not 1000 bytes in a KB, but actually 1024.... apparently some people like to use the term "kibibyte" to refer to 1024 bytes bc KB is so commonly associated with 1000. I now know why my 500 GB external hard drive only has 466 GB :(

Reply:
I don't understand why the units for storage devices are messed up like that, but everywhere else it was always powers of 1024 :)

Reply:
Yeah I was having a conversation with my tech-savvy friend about it, and his theory is that most people would be confused by computer memory being measured in powers of 2 to be very confusing, so they just round it to the nearest thousand, million, etc to avoid explaining why a GB is seemingly arbitrarily decided to be 1,048,576 bytes 😆

Reply:
Ah well, to solve this I'd need to expose all the plugin configuration APIs to Python. Which is a rather big chunk of work, and I kinda hoped that it wouldn't be needed so soon :sweat_smile: I got an idea for a potential workaround, but that didn't work in the end, so you'll have to suffer the warnings for a while until I get to exposing the configuration APIs. It'll happen eventually as a lot other dataset processing tasks need them as well, but not immediately. Sorry.

To explain: some datasets contain images encoded with Basis Universal, which is an image format that's able to transcode itself to GPU-specific block-compression texture formats such as [BCn](https://learn.microsoft.com/en-us/windows/win32/direct3d10/d3d10-graphics-programming-guide-resources-block-compression#bc1) or [ASTC](https://en.wikipedia.org/wiki/Adaptive_scalable_texture_compression). Those are usually about encoding 4x4 blocks of pixels into 64 or 128 bits, thus eight or four times less memory than RGBA with 8 bits per channel. But unfortunately there's no single format all GPUs would support, so one has to explicitly say which format to transcode to, and if it's not done, it decompresses back to RGBA8 that's guaranteed to be supported everywhere. Which however also means that for the Basis images you get four (or eight) times as much data, skewing the stats.

> The script takes a while because it has to print it out for each dataset object

It's not due to the warning, but rather due to the decoding process, which takes time. Regular PNG images also take some time to decode, but Basis images usually contain mip levels so there's much more of them, taking more time, and the RGBA8 decoding code path is probably not well optimized by the library.

Reply:
Ah I see, no problem then, take your time